src/wp-includes/class-wp-user-query.php

```diff
@@ -884,12 +884,15 @@ public function query() {
 				$result->id = $result->ID;
 			}
 		} elseif ( 'all_with_meta' === $qv['fields'] || 'all' === $qv['fields'] ) {
+			$user_ids = array_map( 'intval', $this->results );
```

Comment:
Maybe this is overkill, but since `intval( array( 'foo' ) )` results in `1`, I think it would be preferable to filter out anything that isn't a number.

```suggestion
			$user_ids = array_filter(
				array_map(
					static function ( $user_id ): ?int {
						return is_numeric( $user_id ) ? intval( $user_id ) : null;
					},
					$this->results
				)
			);
```

Also, the above return value of the `users_pre_query` filter is checked for whether or not it is `null` but not whether it is then an `array` as expected. So I think adding that check to be sure it is an array would be good.

Reply:
Revert changes to this class as not needed - https://github.com/WordPress/wordpress-develop/pull/8411/commits/bf504853a1c32444f647db480bef2fa42f4829d6